src/IdentityManager/DataService/IdentityServices/ProfileService.cs

```diff
@@ -26,13 +27,23 @@ public async Task GetProfileDataAsync(ProfileDataRequestContext context)
 			try
 			{
 				var subjectId = context.Subject.GetSubjectId();
-				var user = await _userRepo.FindByIdAsync(Convert.ToInt32(subjectId));
+				var user = await _userRepo.GetAll().Where(x=>x.Id == Convert.ToInt32(subjectId)).FirstOrDefaultAsync();
+
+				//user = new Travely.IdentityManager.Repository.Abstractions.Entities.User
+				//{
+				//	Id = 6,
+				//	Role = Travely.IdentityManager.Repository.Abstractions.Entities.Role.User,
+				//	Agency = new Travely.IdentityManager.Repository.Abstractions.Entities.Agency
+				//	{
+				//		Id = 1
+				//	}
+				//};
 
 				var claims = new List<Claim>
 				{
 					new Claim(JwtClaimTypes.Subject, user.Id.ToString()),
 					new Claim(JwtClaimTypes.Role, user.Role.ToString()),
-					new Claim(JwtClaimTypes.ZoneInfo, user.Agency.ToString())
+					new Claim(JwtClaimTypes.ZoneInfo, user.Agency.Id.ToString())
```

Comment:
I beleive use as AgencyId on it directly. This way you will avoid Agency model redundant materialization.

Reply:
AgencyId does not exist in entities